rfcs/rfc-0004-text-formatting-extensions.md

```diff
@@ -0,0 +1,371 @@
+-   **Font size**: The font size may be too small or too large for some users. This can be mitigated by allowing the user to specify a maximum font size for text sized by the syntax described in this RFC and a minimum font size for text sized by the syntax described in this RFC. However this is up to the client to implement and decide. It is not a requirement of this RFC.
+
+Additionally there may be concerns around the deprecation of KaTeX inline rendering, as some users may have gotten used to it. However, as KaTeX inline rendering is not feasible to implement in non-web clients, it is not considered a choice to keep it. **Not deprecating inline KaTeX rendering would not lead it to suddenly be feasible to implement in non-web clients.**.
+
+# Rationale and alternatives
+
+-   Are there alternative ways to solve this
+
+Probably about a million.
+
+-   Could this be done with existing features or existing solutions
+
+No, Markdown does not support text formatting extensions by default. The deprecated KaTeX inline rendering is not a good fit for text formatting extensions, as it is not designed for general-purpose typesetting and is not feasible to implement in non-web clients.
+
+<!-- - Why is this design the best
+- Are there alternative ways to solve this
+- Could this be done with existing features or existing solutions -->
+
+# Prior art
+
+-   [MFM (Markup language For Misskey)](https://misskey-hub.net/en/docs/for-users/features/mfm). This is considered to not be a good fit for many reasons:
+    -   It is designed for microblogging, not for general-purpose messaging. It is neither consistent with Markdown (in our case RSM) nor is there much use for, for instance, an instant search bar in a messaging app or much of the other fluff that MFM provides.
+-   [BBCode](https://en.wikipedia.org/wiki/BBCode). This is considered to not be a good fit for many reasons:
+    -   Uses a state machine architecture where Markdown traditionally uses an AST. Additionally it does not fit into the existing Markdown (in our case RSM) syntax and is less ergonomic (requires closing tags, see state machine architecture). It is also less expressive and does not allow for the same extensivility.
+
+<!-- This should include both good and bad outlooks on the proposal. This could include how other platforms, software and hardware solve similar issues if relevent or how any existing proposals have tried to solve the same problem. -->
+
+# Unresolved questions
+
+N/A
+
+# Security concerns
+
+As this RFC does not introduce any new security concerns, there are no security concerns to be addressed.
```

Comment:
Only thing is that (at least web) clients should be careful not to use the colour field verbatim, as it could point to an arbitrary URL (in that case of gradient rendering => it uses background to achieve this).